gallery/plot_datapoints.py

```diff
@@ -0,0 +1,122 @@
+"""
+==============
+Datapoints FAQ
+==============
+
+The :mod:`torchvision.datapoints` namespace was introduced to ``torchvision.transforms.v2``. This example showcases what
+these datapoints are and how they behave.
+"""
+
+import PIL.Image
+
+import torch
+import torchvision
+
+# We are using BETA APIs, so we deactivate the associated warning, thereby acknowledging that
+# some APIs may slightly change in the future
+torchvision.disable_beta_transforms_warning()
+
+from torchvision import datapoints
+
+
+########################################################################################################################
+# What are datapoints?
+# --------------------
+#
+# Datapoints are zero-copy tensor subclasses:
+
+tensor = torch.rand(3, 256, 256)
+image = datapoints.Image(tensor)
+
+assert isinstance(image, torch.Tensor)
+assert image.data_ptr() == tensor.data_ptr()
+
+
+########################################################################################################################
+# They are needed in :mod:`torchvision.transforms.v2` for the automatic dispatch to the correct kernel.
+#
+# What datapoints are supported?
+# ------------------------------
+#
+# So far :mod:`torchvision.datapoints` supports four different datapoints:
+#
+# * :class:`~torchvision.datapoints.Image`
+# * :class:`~torchvision.datapoints.Video`
+# * :class:`~torchvision.datapoints.BoundingBox`
+# * :class:`~torchvision.datapoints.Mask`
+#
+# How do I construct a datapoint?
+# -------------------------------
+#
+# Each datapoint class takes any tensor-like data that can be turned into a :class:`~torch.Tensor`
+
+image = datapoints.Image([[0, 1], [1, 0]])
```

Comment:
Let's avoid giving examples with 2d images.
```suggestion
image = datapoints.Image(torch.randint(0, 256, torch.randint(size=(3, 32, 32)))
```
?

Reply:
+1 for not using 2D, but since we print below, I don't want to use (3, 32, 32). I'll just add another dimension, so that we get `(1, 2, 2)`. LMK if that is ok with you.

Reply:
We can maybe just print the shape or few values like `image[0, :5, :5]` instead of everything ?

Reply:
My point is that showing examples with untypical data like 2d images or single channel image as (1, 2, 2) can lead to a certain confusion, like "why they do not use typical examples but some toy stuff". This is how I would see it.

If you would like to have RGB deterministic image, you can do then `torch.arange(3 * 32 * 32).view(3, 32, 32)`
To show the result of inplace addition and multiplication you can just fetch single pixel value:  `image[0, 1, 2]`.

Reply:
> If you would like to have RGB deterministic image, you can do then `torch.arange(3 * 32 * 32).view(3, 32, 32)`

Well that creates int64 data, which is also not terribly realistic, is it?

Reply:
OK OK :) you know how to make it `uint8`, right ? Anyway, I'm not forcing you to change, this is my opinion. Up to you.

Reply:
I've played around with it, and is ok albeit more verbose. However, it somewhat bypasses the point in the comment above the line:

>  Each datapoint class takes any tensor-like data that can be turned into a :class:`~torch.Tensor`

This makes little sense when we just pass a tensor into it as example.